### PR TITLE
Fix py-crypto, tpm2-pytss, tpm2-pkcs11 because openssl isn't building what they expect

### PR DESCRIPTION
**Pull request by bfjelds**
Modify python-cryptography to build rust backend, avoid testing omitted openssl bits.
Modify tpm2-pytss to consume python-cryptography.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Modify python-cryptography to build rust backend, avoid testing omitted openssl bits.
Modify tpm2-pytss to consume python-cryptography.

###### Change Log  
- Modify python-cryptography to build rust backend, avoid testing omitted openssl bits.
- Modify tpm2-pytss to consume python-cryptography.

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=530977&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=532734&view=results

### CODE REVIEW

File: SPECS/tpm2-pkcs11/tpm2-pkcs11.spec [hunk 1]
```diff
@@ -60,10 +60,7 @@ Tools for TCG TPM2 PKCSv11 Software Stack
 %autosetup -p1 -n %{name}-%{version}
 
 %build
-
```
Comment (bfjelds) [inline SPECS/tpm2-pkcs11/tpm2-pkcs11.spec:63]: this isn't really needed

**Review by anphel31 — APPROVED** [review]

File: SPECS/openssl/openssl.spec [hunk 2]
```diff
@@ -162,38 +164,32 @@ export HASHBANGPERL=/usr/bin/perl
...
     enable-rmd160 \
     no-sctp \
-    no-seed \
     no-siphash \
     no-sm2 \
-    no-sm3 \
-    no-sm4 \
     no-ssl \
     no-ssl3 \
+    enable-ssl3-method \
```
Comment (anphel31) [inline SPECS/openssl/openssl.spec:190]: are these needed just to make the python-cryptography tests pass? If so, it actually is probably better to keep openssl as it is, and modify/skip the python-cryptography tests to handle the algorithms we already have.

Comment (bfjelds) [reply SPECS/openssl/openssl.spec:190]: these were the changes i needed to get python-cryptography to build and get the module to load for tests.

we could try to find a way for the rust code not to need them, i didn't have much luck initially, but it is probably possible.

Comment (bfjelds) [reply SPECS/openssl/openssl.spec:190]: i can disable the py-crypto bits that fail without the openssl changes by patching py-crypto.

failures start showing up in tpm2-pytss tests with our existing openssl.  i can probably disable those tests too, though hard to know how important they are (though tpm2-pytss is only needed for swtpm, which is a test tool for us).

Comment (anphel31) [reply SPECS/openssl/openssl.spec:190]: I think that would be best

Comment (bfjelds) [reply SPECS/openssl/openssl.spec]: done ... lots of tests disabled via patches, but everything builds and remaining tests pass :)

**Review by anphel31 — APPROVED** [review]
Can you confirm the buddy build was run on the latest commit?

**Pull request merged**